clang/test/SemaSYCL/sycl-empty-header.cpp

```diff
@@ -0,0 +1,10 @@
+// Test that at least an empty integration header file is created when
+// compiling source files with no device sycl construct.
+
+// RUN: mkdir -p %t_dir
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -save-temps=cwd %s
+// RUN: ls %t.h
+// RUN: rm %t.h
+// RUN: cp %s %t_dir/foo.cpp
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -save-temps=cwd %t_dir/foo.cpp
+// RUN: ls %t.h
```

Comment:
What is the purpose of these lines? Looks like a duplicate of the previous check.

Reply:
There is a subtle difference between when a source file in the cwd is compiled vs when a source file is in a different directory.   For the former, the Driver points to the intermediate files in the TMP directory, and for the latter, the Driver points to them in the cwd.  As part of the former, the call to make the file in the TMP directory also creates it, whereas the latter does not.

I have the two versions to show that in both cases, the header file exists after the integration header pass.

Reply:
According to my understanding LIT runs clang from build directory, whereas `%s` from the first command points to the source directory. If so, there is no difference between these two runs. Am I right?